From 62e201b5c88bfbfbd4c5b6dab6c3554e36e10b64 Fri, 24 Jun 2011 09:09:27 +0200
From: Bram Goffings <bramgoffings@gmail.com>
Date: Fri, 24 Jun 2011 09:09:13 +0200
Subject: [PATCH] attributes optional



diff --git a/includes/form.inc b/includes/form.inc
index 9c2a9c1..53922e6 100644
--- a/includes/form.inc
+++ b/includes/form.inc
@@ -3191,7 +3191,7 @@
       array_unshift($header, '');
     }
   }
-  return theme('table', array('header' => $header, 'rows' => $rows, 'empty' => $element['#empty'], 'attributes' => $element['#attributes']));
+  return theme('table', array('header' => $header, 'rows' => $rows, 'empty' => $element['#empty'], 'attributes' => !empty($element['#attributes']) ? $element['#attributes'] : array()));
 }
 
 /**
